_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,647 @@
+---
+layout: default
+title: Search with k-NN filters
+nav_order: 10
+parent: k-NN
+has_children: false
+has_math: true
+---
+
+# Search for k-NN with filters
+Introduced 2.4
+{: .label .label-purple }
+
+You can create custom filters using Query DSL search options to refine your k-NN searches. You define the filter criteria within the `knn_vector` field's `filter` subsection in your query. The most commonly used Query DSL query types are: term, range, regex and wildcard. To include or exclude results, you specify Boolean query clauses. You also specify a query point with the `knn_vector` type and search for nearest neighbors that match your filter criteria.
```

Comment:
`The most commonly used Query DSL query types are: term, range, regex and wildcard.`
I think we better change this to point out that any of the registered query types can be used. Maybe something like:
`You can use any of DSL query types that are registered in the OpenSearch as a filter. This includes but not limited to existing OpenSearch types like term, range, regex or wildcard, as well as custom query types`

Reply:
main idea is that we do support all standard types including most popular ones, but also any other custom one

Reply:
ok i can change that.